cms/djangoapps/contentstore/views/component.py

```diff
@@ -154,7 +154,11 @@ def container_handler(request, usage_key_string):
     """
     if 'text/html' in request.META.get('HTTP_ACCEPT', 'text/html'):
 
-        usage_key = UsageKey.from_string(usage_key_string)
+        try:
+           usage_key = UsageKey.from_string(usage_key_string)
+        except Exception:
```

Comment:
Always try to catch specific exception not generic exception. For example you can catch 'InvalidKeyError' exception here.